packages/lisk-p2p/src/p2p.ts

```diff
@@ -250,35 +250,30 @@ export class P2P extends EventEmitter {
 		await this._stopHTTPServer();
 	}
 
-	private _connectToPeers(): void {
-		const availablePeers = Array.from(this._newPeers).map(
-			(peerInfo: PeerInfo) => new Peer(peerInfo),
-		);
+	private _connectToPeers(peers: ReadonlyArray<PeerInfo>): void {
```

Comment:
The `this._peerPool.connectToPeers(...)` seems like an implementation detail which could be part of the `this._peerPool.runDiscovery(...)` method. Also, the name `connectToPeers` is confusing because the PeerPool will only connect to some of the peers we provide (not all of them).

Maybe there is a simpler way to populate the `_newPeers` and `_triedPeers` maps:

- We can emit new kinds of events on the PeerPool and then listen to those events from the P2P class to populate the _newPeers and _triedPeers?
- Or we can move the _newPeers and _triedPeers maps into the PeerPool (and expose them to the P2P module via getter functions)?

Reply:
We don't want to make `this._peerPool.connectToPeers(...)` part of `this._peerPool.runDiscovery(...)` because we still want to keep them independent as they could be called separately as well. For current protocol, of course, it runs discovery and `connectToPeers` together on startup of a node but later when we will implement lip, it doesn't necessary means that we will always run connectToPeers after discovery or we might want to call  `connectToPeers` separately without running discovery, for example, in periodic shuffling(it doesn't involve discovery and it connects with peers randomly).

Reply:
@ishantiw When we introduce shuffling, we could add a new `peerPool.shufflePeers()` method? Shuffling will also involve disconnecting from old peers; this could be handled internally by the peer pool; `peerPool.shufflePeers()`.

My concern is also partly related to the name of the method `peerPool.connectToPeers(peers)`. It sounds like it will connect to all provided peers which is not the case. If the name of the method was accurate, it should be called `peerPool.runPeerSelectionAndConnection(peers)` (or something similar); that way it sounds like the same level of abstraction as `runDiscovery()`.

Reply:
actually after merging #1030  we don't really need `p2p._connectToPeers()` because all we are doing in this method is run the method `connectToPeers()` of `PeerPool` and getting a list of `PeerInfo` in return which was chosen to connect but we are not actually sure that we connected. So I will remove this function in P2P and add the statement to run `peerPool.connectToPeers` rename to `peerPool.selectPeersToConnect` in p2p.Start(), like you said that we need a proper name. 

Yes, for shuffling I think we will do it `PeerPool` in a separate handler. But still, we need to notify p2p to manage peer bucket. I guess it could become easier if we could have buckets in `PeerPool` 🤔

Reply:
👍 `peerPool.selectPeersToConnect` sounds good.